src/librustc_errors/emitter.rs

```diff
@@ -724,7 +724,10 @@ impl EmitterWriter {
         }
         match write!(&mut self.dst, "\n") {
             Err(e) => panic!("failed to emit error: {}", e),
-            _ => ()
+            _ => match self.dst.flush() {
```

Comment:
This flush allows us to print the last "\n" at the end of the errors printing:

```
error[E0101]: cannot determine a type for this expression: unconstrained type
  --> src/test/compile-fail/E0101.rs:12:13
   |
12 |     let x = |_| {};
   |             ^^^^^^ cannot resolve type of expression

error: aborting due to previous error
jturner-23759: rust jturner$
```

vs

```
error[E0101]: cannot determine a type for this expression: unconstrained type
  --> src/test/compile-fail/E0101.rs:12:13
   |
12 |     let x = |_| {};
   |             ^^^^^^ cannot resolve type of expression

error: aborting due to previous error

jturner-23759: rust jturner$
```